node/basicnode/string.go

```diff
@@ -13,8 +13,12 @@ var (
 )
 
 func NewString(value string) datamodel.Node {
-	v := plainString(value)
-	return &v
+	return plainString(value)
+}
+
+// NewStringAddr is like NewString, but it does not allocate a new string.
```

Comment:
A short comment here about how this lets you make a mutable node, and warning that that might not be what you or others expect, would make me happier.